File: santander app/app/src/main/java/com/desafiosantander/DashBoard/DashBoardConfigurator.java
package com.desafiosantander.DashBoard;

import java.lang.ref.WeakReference;


public enum DashBoardConfigurator {
    INSTANCE;

    public void configure(DashBoardActivity activity) {

        DashBoardRouter router = new DashBoardRouter();
        router.activity = new WeakReference<>(activity);

        DashBoardPresenter presenter = new DashBoardPresenter();
        presenter.output = new WeakReference<DashBoardActivityInput>(activity);

        DashBoardInteractor interactor = new DashBoardInteractor();
        interactor.output = presenter;

        if (activity.output == null) {
            activity.output = interactor;
        }
        if (activity.router == null) {
            activity.router = router;
        }
    }
}

File: BankApp/app/src/main/java/com/accenture/bankapp/screens/dashboard/DashboardConfigurator.kt
package com.accenture.bankapp.screens.dashboard

import timber.log.Timber
import java.lang.ref.WeakReference

object DashboardConfigurator {
    fun configureFragment(dashboardFragment: DashboardFragment) {
        Timber.i("configureFragment: Configuring the Dashboard Fragment")

        val dashboardRouter = DashboardRouter()
        val dashboardPresenter = DashboardPresenter()
        val dashboardInteractor = DashboardInteractor()

        dashboardRouter.dashboardFragment = WeakReference(dashboardFragment)
        dashboardFragment.dashboardRouter = dashboardRouter

        dashboardPresenter.dashboardFragmentInput = WeakReference(dashboardFragment)

        dashboardInteractor.dashboardFragment = WeakReference(dashboardFragment)
        dashboardInteractor.dashboardFragmentInput = WeakReference(dashboardFragment)
        dashboardInteractor.dashboardPresenterInput = dashboardPresenter

        dashboardFragment.dashboardRouter = dashboardRouter
        dashboardFragment.dashboardInteractorInput = dashboardInteractor
    }
}